back-end/hub-core/src/main/java/io/apicurio/hub/core/integrations/EventsServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.apicurio.hub.core.integrations;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @author Fabian Martinez
+ */
+@ApplicationScoped
+public class EventsServiceImpl implements EventsService {
+
+    private static final Logger log = LoggerFactory.getLogger(EventsServiceImpl.class);
+
+    private ObjectMapper mapper;
+    private boolean configuredSinks = false;
+    private List<EventSink> subscriptions;
+    private ExecutorService executor;
+
+    @Inject
+    Instance<EventSink> sinks;
+
+    @PostConstruct
+    public void init() {
+        for (EventSink sink : sinks) {
+            if (sink.isConfigured()) {
+                configuredSinks = true;
+                if (subscriptions == null) {
+                    subscriptions = new ArrayList<>();
+                }
+                log.info("Subscribing sink " + sink.name());
+                subscriptions.add(sink);
+            }
+        }
+        if (configuredSinks) {
+            executor = new ThreadPoolExecutor(1, 3, 5, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>());
+        }
+    }
+
+    @Override
+    public boolean isConfigured() {
+        return configuredSinks;
+    }
+
+    @Override
+    public void triggerEvent(ApicurioEventType type, String designId, Object data) {
+        if (configuredSinks && data != null) {
+            byte[] buffer;
+            try {
+                buffer = getMapper().writeValueAsBytes(data);
+            } catch (JsonProcessingException e) {
+                log.error("Error serializing event data", e);
+                return;
+            }
+            executor.submit(() -> {
+               for (EventSink sink : subscriptions) {
+                   try {
+                       sink.handle(type, designId, buffer);
+                   } catch (Exception e) {
+                       log.error("Error sending event in sink " + sink.name(), e);
+                   }
+               }
+            });
+        }
+    }
+
+    private synchronized ObjectMapper getMapper() {
+        if (mapper == null) {
+            mapper = new ObjectMapper();
+            mapper.setSerializationInclusion(Include.NON_NULL);
+        }
```

Comment:
Maybe initialize the mapper in the `init()` method to remove this synchronized method?  Or if you prefer the lazy init approach, I suggest you remove `synchronized` from the method.  Because who cares if you actually create more than 1 due to the race condition?  In the case of `ObjectMapper` that doesn't really matter.  It's not worth the cost of synchronization IMO.

Reply:
ok, I'll remove the `synchronized` even knowing there is a possible race condition